src/components/tile/tile-content/tile-content.style.ts

```diff
@@ -0,0 +1,73 @@
+import styled, { css } from "styled-components";
+import { space } from "styled-system";
+import baseTheme from "../../../style/themes/base";
+import computeSizing from "../../../style/utils/element-sizing";
+import { TileContentProps } from "./tile-content.component";
+
+interface StyledTileContentProps extends TileContentProps {
+  isHorizontal?: boolean;
+  isVertical?: boolean;
+  width?: string | number;
+  height?: string | number;
+}
+
+const StyledTileContent = styled.div<StyledTileContentProps>`
+  ${({ isHorizontal, isVertical, width, height }) => css`
+    ${space}
+
+    box-sizing: border-box;
+    position: relative;
+    flex-grow: 1;
+
+    ${isHorizontal &&
+    css`
+      display: inline;
+
+      :last-of-type {
+        padding-right: 0;
+      }
+
+      :first-of-type {
+        padding-left: 0;
+      }
+
+      & + & {
+        margin-top: 0;
+        border-left: solid 1px var(--colorsUtilityMajor050);
+      }
+    `}
+
+    ${isVertical &&
+    css`
+      width: auto;
+
+      :last-of-type {
+        padding-bottom: 0;
+      }
+
+      :first-of-type {
+        padding-top: 0;
+      }
+
+      & + & {
+        margin-top: 0;
+        border-top: solid 1px var(--colorsUtilityMajor050);
+      }
+    `}
+
+    ${(width || height) &&
+    css`
+      flex-grow: 0;
+      ${computeSizing({
+        width: width || /* istanbul ignore next */ undefined,
```

Comment:
**suggestion/question:** What was the reason for including the `undefined` or ignoring this? I've checked the `default` example in Storybook whilst console logging the width value for this component and it returns `undefined`. If there is a genuine reason that's all good.

![Screenshot 2023-10-09 at 11 25 29](https://github.com/Sage/carbon/assets/56251247/55a440d0-5efe-4bc4-bbf7-7016c759ca4f)

Reply:
This is just maintaining the existing implementation that ignores when the width is set to 0 so that it fits the content etc